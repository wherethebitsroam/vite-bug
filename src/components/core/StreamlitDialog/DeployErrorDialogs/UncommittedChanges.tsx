import React from "react"
import { IDeployErrorDialog } from "./types"

function UncommittedChanges(repository: string): IDeployErrorDialog {
  const repoName = repository
    .split("/")
    .slice(1)
    .join("/")

  return {
    title: "Unable to deploy app",
    body: (
      <>
        <p>
          Your repository <code>{repoName}</code> has uncommitted changes.
        </p>
        <p>Please commit the latest changes and push to GitHub to continue.</p>
      </>
    ),
  }
}

export default UncommittedChanges
