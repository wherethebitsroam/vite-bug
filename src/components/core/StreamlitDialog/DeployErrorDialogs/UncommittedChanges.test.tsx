import UncommittedChanges from "./UncommittedChanges"

describe("DeployErrorDialog - RepoIsAhead", () => {
  it("should render without crashing", () => {
    const dialog = UncommittedChanges("user/reponame")

    expect(dialog).toMatchSnapshot()
  })
})
