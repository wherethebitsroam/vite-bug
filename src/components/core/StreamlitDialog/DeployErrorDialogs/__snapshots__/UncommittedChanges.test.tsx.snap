// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeployErrorDialog - RepoIsAhead should render without crashing 1`] = `
Object {
  "body": <React.Fragment>
    <p>
      Your repository 
      <code>
        reponame
      </code>
       has uncommitted changes.
    </p>
    <p>
      Please commit the latest changes and push to GitHub to continue.
    </p>
  </React.Fragment>,
  "title": "Unable to deploy app",
}
`;
